File: MgDev/UnitTest/WebTier/DotNet/TestCommon/ResourceService/ResourceServiceOperationExecutor.cs
using OSGeo.MapGuide.Test.Common;
using SqliteDotNet;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OSGeo.MapGuide.Test.Operations
{
    /// <summary>
    /// The base class of all test executors using the MgResourceService
    /// </summary>
    /// <typeparam name="T">The test executor subclass type</typeparam>
    public abstract class ResourceServiceOperationExecutor<T> : PlatformApiTestExecutor
    {
        protected MgResourceService _resourceService;

        protected ResourceServiceOperationExecutor(MgResourceService resSvc, string unitTestVm)
            : base(typeof(T).Name.ToUpper(), ApiTypes.Platform, unitTestVm)
        {
            _resourceService = resSvc;
        }
    }
}

File: MgDev/UnitTest/WebTier/Java/src/org/osgeo/mapguide/test/platform/operations/resource/ResourceServiceOperationExecutor.java
package org.osgeo.mapguide.test.platform.operations.resource;

import org.osgeo.mapguide.*;
import org.osgeo.mapguide.test.*;
import org.osgeo.mapguide.test.common.*;

import com.almworks.sqlite4java.*;

public abstract class ResourceServiceOperationExecutor extends PlatformApiTestExecutor
{
    protected MgResourceService _resourceService;

    protected ResourceServiceOperationExecutor(String opName, MgResourceService resSvc, String unitTestVm)
    {
        super(opName, ApiTypes.Platform, unitTestVm);
        _resourceService = resSvc;
    }
}